Task11

VadymSalymonenko opened the pull request:
added code and Cmake for task 11.

VadymT95 committed:
added task1 code and Cmake file.

VadymT95 committed:
added code and Cmake for task 2.

VadymT95 committed:
added code and Cmake for task 3.

VadymT95 committed:
added code and Cmake for task 4.

VadymT95 committed:
added code and Cmake for task 6.

VadymT95 committed:
added code and Cmake for task 7.

VadymT95 committed:
added code and Cmake for task 8.

VadymT95 committed:
added code and Cmake for task 9.

VadymT95 committed:
added code and Cmake for task 10.

VadymT95 committed:
added code and Cmake for task 11.

BBuinich reviewed (changes requested):
looks good to me, leave only task 11

**Consistency:** I notice that sometimes you use long long and sometimes you use int for storing worry levels and other values. We should ensure consistency throughout.
**Code Comments:** While you have provided comments, they can be more descriptive for better clarity.
**Refactoring:** Some parts of the code, like the process of reading from the file and then processing monkeys, can be split into smaller functions for clarity.
**Error Checks:** It's a good idea to add more error checks, especially when dealing with file I/O and dynamic memory allocations.

VadymT95 committed:
Code optimization.

1) added comments to the main logic;
2) fill_monkeys_from_file function split into smaller ones.

BBuinich reviewed (approved):
Merge it
